### PR TITLE
optional_plugins: Increase test coverage of mux-suite

### PR DESCRIPTION
**Pull request by ldoktor**
Add tests for other yaml-loader features that I use in my CI (and want them to stay functional :-)). While on it also added codecov config to not to require CI PASS.

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+codecov:
+  notify:
+    require_ci_to_pass: no
```
Comment (clebergnu) [inline codecov.yml]: Do you know what would happen if there was a major failure, and say, just the very first test was executed?  Would that bring the coverage numbers down?

Comment (clebergnu) [reply codecov.yml]: @ldoktor had a question here for you.

Comment (ldoktor) [reply codecov.yml]: IIUC it has nothing to do with the coverage run, but with the rest of github checks. I noticed the Codecov report only after all checks are successfully performed.

Comment (ldoktor) [reply codecov.yml]: ... as for the coverage, it's up to us. We call the "upload" results so we can decide whether to do it on incomplete runs or not.

**Pull request merged**